Doc guide for custom domains (#852)
openziti/zrok: docs/guides/custom-domains/index.mdx

Reviewer: My personal style preference would be fewer enthusiastic `!` bangs.

---
title: Custom Domains
---

## Overview
[myzrok.io](https://myzrok.io) is a hosted zrok-as-a-service offering that provides a way for you bring a custom DNS name for zrok shares.
For example, let's say you own the domain `foo.example.io`, you can leverage zrok custom domains to
create ephemeral shares such as: `https://vw8jbg4ijz5g.foo.example.io`
or [reserved shares](/concepts/sharing-reserved.md) such as `https://myshare.foo.example.io`.

Custom domains require a Pro subscription with [myzrok.io](https://myzrok.io).
If you don't already have an account, you can sign up for one [here](https://myzrok.io).

[myzrok.io](https://myzrok.io) provides a guided setup with just a few easy steps!

1. Bring your own custom domain name

> 2. Create DNS validation records and a CNAME record

Reviewer: ```suggestion
2. Create DNS validation CNAME-record for certificate management and A-record round-robin for the frontend
```

Reviewer: The `-record` suffixes might be too noisy, but I thought it best to avoid implying the A records are for "validation," when only the CNAME record is for validation, i.e., delegating cert mgmt authority, and the A records are a round robin for the frontend proxy.

Author: round-robin is a load balancer thing, not a DNS thing, so I'd prefer not to muddy the waters with that term here. How about something more like "Create DNS records for certificate validation and traffic routing" (leave it generic and the specifics will get covered below).

Author: Ah, I stand corrected and did not know that. My bad

3. Wait for zrok to validate your records and finalize configuration
4. Start sharing!

Detailed setup instructions are documented below.

### Prerequisites
:::note
In order to create a custom domain in zrok, **you must already own the domain you want to use.**
:::

> During the setup process you will need to create DNS records to validate ownership and to allow a certiicate to be issued

Reviewer: ```suggestion
During the setup process, you will need to create DNS records to validate ownership and allow a certificate to be issued
```

on behalf of your domain. Once you have your domain registered, you can begin the process of setting up your custom
domain with zrok!

### Create Your Custom Domain

> Log into the myzrok Console and access the domains page by clicking on the globe icon in the left navigation menu.

Reviewer: ```suggestion
Log into the myzrok.io console and access the domains page by clicking on the globe icon in the left navigation menu.
```

Caps for a reason?

myzrok.io for consistency

![myzrok_domains_page](images/myzrok_domains_page.png)

Custom domains require a Pro subscription in [myzrok.io](https://myzrok.io)

Click the CREATE button on the top right of the page to get started.
When you click the create button you’ll be presented with a form to allow you to enable your custom domain.
Enter your domain into the form field and click CREATE. This will begin the process for setting up your custom domain.
A new managed TLS certificate will be created to host traffic on your domain's behalf.

![myzrok_add_domain](images/myzrok_add_domain.png)

This may take a few minutes.
You may close the form at this time and come back when your domain is *pending validation.*

> Once your certificate is ready, you’ll be presented with instructions on how to set up your DNS records.

Reviewer: A rough estimate for timing will be appreciated. At a minimum, how long to wait before raising a support request.

Author: We don't really have control over this, and AWS doesn't give estimates on it either

### Creating DNS Records

zrok will host and manage a TLS certificate for the custom domain on your behalf.
This process requires a DNS validation record to be created in order to prove ownership of the domain.
Follow the prompts in the UI to create a CNAME DNS record with the name and value specified in the UI.

![myzrok_add_cname](images/myzrok_add_cname.png)

Next, create an A record to direct all DNS requests for your domain to a set of static IPs that are hosted by zrok.

![myzrok_add_a_record](images/myzrok_add_a_record.png)

> After you’ve created your records, you can verify that they are configured properly using the instructions provided in the form.

Reviewer: Which specific details _must_ match for me to know I did it right?

![myzrok_verify_dns](images/myzrok_verify_dns.png)

> Once you have created your DNS records, it will take zrok a few minutes to validate that they exist.

Reviewer: How long is too long and I should contact support?

You can safely close the form until your certificate has been issued.

### Finalizing Your Custom Domain

> After your records have been validated and your certificate has been issued, click the Finalize button within 72 hours to complete your custom domain setup.

Reviewer: ```suggestion
After your records have been validated and your certificate has been issued, click the FINALIZE button within 72 hours to complete your custom domain setup.
```

consistent caps for buttons

![myzrok_finalize](images/myzrok_finalize.png)

From here, myzrok.io will complete the last few steps of creating your custom domain.
This should only take a minute, but if you need to close the form you can find the instructions on how to share your frontend when you return.

### Start Sharing!

> Once the Finalize stage has completed, you can start sharing with your custom DNS!

Reviewer: ```suggestion
Once the finalize stage has completed, you can start sharing with your custom DNS!
```

Not referencing a button or proper noun, so no caps.
